src/datasets/infra/repositories/transformers/datasetTransformers.ts

```diff
@@ -58,49 +69,64 @@ const transformPayloadToDatasetLicense = (licensePayload: any): DatasetLicense =
   }
   return datasetLicense;
 };
-// eslint-disable-next-line  @typescript-eslint/no-explicit-any
-const transformPayloadToDatasetMetadataBlocks = (metadataBlocksPayload: any): DatasetMetadataBlocks => {
+
+const transformPayloadToDatasetMetadataBlocks = (
+  metadataBlocksPayload: MetadataBlocksPayload,
+): DatasetMetadataBlocks => {
   return Object.keys(metadataBlocksPayload).map((metadataBlockKey) => {
+    const metadataBlock = metadataBlocksPayload[metadataBlockKey];
     return {
-      name: metadataBlockKey,
-      fields: transformPayloadToDatasetMetadataFields(metadataBlocksPayload[metadataBlockKey].fields),
+      name: metadataBlock.name,
+      fields: transformPayloadToDatasetMetadataFields(metadataBlock.fields),
     };
   }) as DatasetMetadataBlocks;
 };
-// eslint-disable-next-line  @typescript-eslint/no-explicit-any
-const transformPayloadToDatasetMetadataFields = (metadataFieldsPayload: any): DatasetMetadataFields => {
-  const metadataFieldKeys = Object.keys(metadataFieldsPayload);
-  const metadataFields: DatasetMetadataFields = {};
-  for (const metadataFieldKey of metadataFieldKeys) {
-    const metadataField = metadataFieldsPayload[metadataFieldKey];
-    const metadataFieldTypeName = metadataField.typeName;
-    metadataFields[metadataFieldTypeName] = transformPayloadToDatasetMetadataFieldValue(metadataField.value);
-  }
-  return metadataFields;
+
+const transformPayloadToDatasetMetadataFields = (
+  metadataFieldsPayload: MetadataFieldPayload[],
+): DatasetMetadataFields => {
+  return metadataFieldsPayload.reduce((acc: DatasetMetadataFields, field: MetadataFieldPayload) => {
```

Comment:
Minor detail - Can we rename "acc" to something more understandable? Like `datasetMetadataFieldsResult` or simply `result`